kythe/typescript/indexer.ts

```diff
@@ -32,6 +32,24 @@ export interface VName {
   language: string;
 }
 
+/**
+ * A indexer plugin adds extra functionality with the same inputs as the base
+ * indexer.
+ */
+export interface Plugin {
+  name: string;
+  index:
+      (/**
+        * The VName for the CompilationUnit, containing compilation-wide info.
+        */
+       compilationUnit: VName,
```

Comment:
I don't see a need to pass this to the plugins anymore.

Reply:
if in particular, the path argument to pathVNames could be undefined.

Reply:
I think it's still needed; the `compilationUnit` contains the root and corpus of the project. We could get rid of `pathVNames` though, since that defers to `compilationUnit` if the path doesn't exist (per the existing model of the TS indexer):

https://github.com/kythe/kythe/blob/16bf176d3eb7d0860eed4c9c6785674470a82763/kythe/typescript/indexer.ts#L168-L178

Reply:
True, it's needed in newFileVName, but I don't see why it needs to be passed to each plugin. In particular, if you instead had
`pathVNames: (path: string?) => VName | undefined, paths: string[]`
then passing undefined would be equivalent to querying for the compilation unit.

Reply:
Okay, I see what you're saying. I agree, and I think it would be even better just to pass `newFileVName`, since the compilation unit's only use is to make path vnames.